src/Adapter/Image/Uploader/ProductImageDestinationFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use Image;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject\ImageId;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+
+//@todo does it need interface?
```

Comment:
In theory, if It's in adapter namespace, yes, because one day we will rewrite it with another implementation.

But it's a good opportunity to check this rule still applies. @eternoendless one day you told me you thought we create too much interfaces. Is this a good usecase ?

Reply:
I've changed the implementation, this class is no longer needed.

Reply:
In general adapters need an interface because they are meant to be replaced by core reimplementations. Ideally, these interfaces should be well thought and not just mimic the legacy class. You can also rely on interface segregation for that (aka "don't make clients depend on functionality they don't need" aka "make your interfaces simple" aka "implement more than one interface if you need to")